README.md

```diff
@@ -56,130 +59,121 @@ Expoes an interface to AES encryption (formerly Rijndael), as defined in U.S. Fe
-          // prints: Hello, world!
-        }
-      })
-    }
-  })
+  const decipher = await crypto.aes.create(key128, IV)
+  const decryptedBuffer = await cipher.decrypt(encryptedBuffer)
+
+  console.log(decryptedBuffer)
+  // prints: <Buffer 42 f1 67 d9 2e 42 d0 32 9e b1 f8 3c>
+
+  console.log(decryptedBuffer.toString('utf-8'))
+  // prints: Hello, world!
 }
-main()
 
+main()
 ```
 
 ### `crypto.hmac`
 
 Exposes an interface to the Keyed-Hash Message Authentication Code (HMAC) as defined in U.S. Federal Information Processing Standards Publication 198. An HMAC is a cryptographic hash that uses a key to sign a message. The receiver verifies the hash by recomputing it using the same key.
 
-#### `crypto.hmac.create(hash, secret, callback)`
+#### `crypto.hmac.create(hash, secret)`
 
 - `hash: String`
 - `secret: Buffer`
-- `callback: Function`
 
-##### `digest(data, callback)`
+Returns `Promise<{digest<Function>}>`
+
+##### `digest(data)`
 
 - `data: Buffer`
-- `callback: Function`
+
+Returns `Promise<Buffer>`
 
 Example:
 
 ```js
-var crypto = require('libp2p-crypto')
+const crypto = require('libp2p-crypto')
 
-let hash = 'SHA1' // 'SHA256' || 'SHA512'
+async function main () {
+  const hash = 'SHA1' // 'SHA256' || 'SHA512'
+  const hmac = await crypto.hmac.create(hash, Buffer.from('secret'))
+  const sig = await hmac.digest(Buffer.from('hello world'))
+  console.log(sig)
+}
 
-crypto.hmac.create(hash, Buffer.from('secret'), (err, hmac) => {
-  if (!err) {
-    hmac.digest(Buffer.from('hello world'), (err, sig) => {
-      if (!err) {
-        console.log(sig)
-      }
-    })
-  }
-})
+main()
 ```
 
 ### `crypto.keys`
 
 **Supported Key Types**
 
-The [`generateKeyPair`](#cryptokeysgeneratekeypairtype-bits-callback), [`marshalPublicKey`](#cryptokeysmarshalpublickeykey-type-callback), and [`marshalPrivateKey`](#cryptokeysmarshalprivatekeykey-type) functions accept a string `type` argument.
+The [`generateKeyPair`](#generatekeypairtype-bits), [`marshalPublicKey`](#marshalpublickeykey-type), and [`marshalPrivateKey`](#marshalprivatekeykey-type) functions accept a string `type` argument.
 
 Currently the `'RSA'` and `'ed25519'` types are supported, although ed25519 keys support only signing and verification of messages.  For encryption / decryption support, RSA keys should be used.
 
 Installing the [libp2p-crypto-secp256k1](https://github.com/libp2p/js-libp2p-crypto-secp256k1) module adds support for the `'secp256k1'` type, which supports ECDSA signatures using the secp256k1 elliptic curve popularized by Bitcoin.  This module is not installed by default, and should be explicitly depended on if your project requires secp256k1 support.
 
-### `crypto.keys.generateKeyPair(type, bits, callback)`
+### `crypto.keys.generateKeyPair(type, bits)`
 
 - `type: String`, see [Supported Key Types](#supported-key-types) above.
 - `bits: Number` Minimum of 1024
-- `callback: Function`
+
+Returns `Promise<privateKey<Buffer>, publicKey<Buffer>>`
```

Comment:
Does the Promise resolve to an array or an object?

Reply:
Lol comments as a type system.

I think it's supposed to be an object with `privateKey` and `publicKey` properties, both of which are buffers.

Reply:
Will update the docs

Reply:
Done!